game-core/src/main/java/games/strategy/debug/Console.java

```diff
@@ -0,0 +1,100 @@
+package games.strategy.debug;
+
+import java.awt.BorderLayout;
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+
+import javax.swing.Action;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JToolBar;
+import javax.swing.SwingConstants;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+import games.strategy.engine.framework.lookandfeel.LookAndFeelSwingFrameListener;
+import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.ui.SwingAction;
+import games.strategy.ui.SwingComponents;
+import lombok.extern.java.Log;
+import swinglib.JComboBoxBuilder;
+
+/**
+ * A 'console' window to display log messages to users.
+ */
+@Log
+public class Console {
+  private final JTextArea textArea = new JTextArea(20, 50);
+
+  private final JFrame frame = new JFrame("TripleA Console");
+
+  public Console() {
+    final Level logLevel = ClientSetting.LOGGING_VERBOSITY.value().equals(Level.ALL)
+        ? Level.INFO
+        : Level.WARNING;
+    LogManager.getLogManager().getLogger("").setLevel(logLevel);
+
+    ClientSetting.SHOW_CONSOLE.addSaveListener(newValue -> {
+      if (newValue.equals(String.valueOf(true))) {
+        SwingUtilities.invokeLater(() -> setVisible(true));
+      }
+    });
+
+    SwingComponents.addWindowClosedListener(frame, () -> ClientSetting.SHOW_CONSOLE.saveAndFlush(false));
+    LookAndFeelSwingFrameListener.register(frame);
+    frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+    frame.getContentPane().setLayout(new BorderLayout());
+    textArea.setLineWrap(true);
+    textArea.setWrapStyleWord(true);
+    frame.getContentPane().add(new JScrollPane(textArea), BorderLayout.CENTER);
+    final JToolBar actions = new JToolBar(SwingConstants.HORIZONTAL);
+    frame.getContentPane().add(actions, BorderLayout.SOUTH);
+    actions.setFloatable(false);
+    final Action threadDiagnoseAction =
+        SwingAction.of("Enumerate Threads", e -> log.info(DebugUtils.getThreadDumps()));
```

Comment:
I was actually tempted to raise an issue to drop this. Presumably a thread dump is useful when there is a deadlock, if we have that then you are not going to be able to click this button. I suspect it may have been here in part to verify the early multi-threaded code, back in Java 1.3 and 1.4 that was more difficult (and at the time the dev's were learning too).